Projects/Playground/Playground.Uwp/App.xaml.cs

```diff
@@ -1,107 +1,19 @@
-        {
-            var rootFrame = Window.Current.Content as Frame;
-
-            // Do not repeat app initialization when the Window already has content,
-            // just ensure that the window is active
-            if (rootFrame == null)
-            {
-                // Create a Frame to act as the navigation context and navigate to the first page
-                rootFrame = new Frame();
-
-                rootFrame.NavigationFailed += OnNavigationFailed;
-
-                if (e.PreviousExecutionState == ApplicationExecutionState.Terminated)
-                {
-                    //TODO: Load state from previously suspended application
-                }
-
-                // Place the frame in the current Window
-                Window.Current.Content = rootFrame;
-            }
-
-            if (e.PrelaunchActivated == false)
-            {
-                if (rootFrame.Content == null)
-                {
-                    var setup = new Setup(rootFrame, "Suspension");
-                    setup.Initialize();
-
-                    var start = Mvx.Resolve<IMvxAppStart>();
-                    start.Start();
-                }
-
-                // Ensure the current window is active
-                Window.Current.Activate();
-            }
-        }
-
-        /// <summary>
-        /// Invoked when Navigation to a certain page fails
-        /// </summary>
-        /// <param name="sender">The Frame which failed navigation</param>
-        /// <param name="e">Details about the navigation failure</param>
-        void OnNavigationFailed(object sender, NavigationFailedEventArgs e)
-        {
-            throw new Exception("Failed to load Page " + e.SourcePageType.FullName);
-        }
-
-        /// <summary>
-        /// Invoked when application execution is being suspended.  Application state is saved
-        /// without knowing whether the application will be terminated or resumed with the contents
-        /// of memory still intact.
-        /// </summary>
-        /// <param name="sender">The source of the suspend request.</param>
-        /// <param name="e">Details about the suspend request.</param>
-        private async void OnSuspending(object sender, SuspendingEventArgs e)
-        {
-            var deferral = e.SuspendingOperation.GetDeferral();
-
-            var suspension = Mvx.GetSingleton<IMvxSuspensionManager>() as MvxSuspensionManager;
-
-            await suspension.SaveAsync();
-            //TODO: Save application state and stop any background activity
-            deferral.Complete();
-        }
-
-        private async void OnResuming(object sender, object e)
+        protected override MvxWindowsSetup CreateSetup(Frame rootFrame, string suspension)
         {
-            var suspension = Mvx.GetSingleton<IMvxSuspensionManager>() as MvxSuspensionManager;
-            await suspension.RestoreAsync();
+            return new MvxTypedWindowsSetup<Core.App>(rootFrame, suspension);
```

Comment:
The setup is normally automatically called from somewhere if i remember correctly. Why does this need to be created here if that is the case?